src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp

```diff
@@ -54,38 +54,44 @@ bool AnalogServerHandler::_handleIAnalog(yarp::os::Bottle &cmd, yarp::os::Bottle
     if (is==nullptr)
       return false;
 
-    int msgsize=cmd.size();
+    const int msgsize=cmd.size();
+    int ret = IAnalogSensor::AS_ERROR;
 
     int code=cmd.get(1).asVocab();
     switch (code)
     {
     case VOCAB_CALIBRATE:
       if (msgsize==2)
-        is->calibrateSensor();
+        ret = is->calibrateSensor();
       else
       {
-        //read Vector of values and pass to is->calibrate();
+        Vector v(msgsize - 2);
```

Comment:
What if  `msgsize` is 0 or 1? This initialization seems risky
And I would name that `2`, for code readability.

Reply:
> What if `msgsize` is 0 or 1? This initialization seems risky

Done at https://github.com/robotology/yarp/pull/2003/commits/a4909c55c1aedf11575f53fb2bb54f1c41ae7878.

> And I would name that `2`, for code readability.

I'm sorry, what do you mean?

Reply:
> I'm sorry, what do you mean? 

I'd use a variable like e.g. `size_t offset = 2;` and then ` Vector v(msgsize - offset);`.
Not sure if it is actually an offset, using a variable make the code more readable, but it is a personal taste

Reply:
Done at https://github.com/robotology/yarp/pull/2003/commits/776bc3feaac113e11266fb8fddf03ca77e6b40e3.